### PR TITLE
Stop battle camera always following

### PR DESCRIPTION
**Pull request by PhilPJL**
Stops the battle camera following every time it switches character.

Fixes https://github.com/ChrisPJohn/SolastaCommunityExpansion/issues/125

### CODE REVIEW

File: SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs [hunk 1]
```diff
@@ -94,6 +94,22 @@ internal static void DisplayGameUi()
                 Main.Settings.PermanentSpeedUp = toggle;
             }
 
+            toggle = Main.Settings.DontFollowCharacterInBattle;
+
+            if (UI.Toggle("Battle camera doesn't follow when character is already on screen.", ref toggle, UI.AutoWidth()))
```
Comment (ThyWoof) [inline SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs:99]: we don't use any final punctuation on UI toggles

File: SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs [hunk 2]
```diff
@@ -94,6 +94,22 @@ internal static void DisplayGameUi()
...
+            {
+                Main.Settings.DontFollowCharacterInBattle = toggle;
+            }
+
+            if (Main.Settings.DontFollowCharacterInBattle)
+            {
+                intValue = Main.Settings.DontFollowMargin;
+                if (UI.Slider("Unless character is off or within % of screen edge.".italic().white(), ref intValue, 0, 15, 1, "%", UI.AutoWidth()))
```
Comment (ThyWoof) [inline SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs:107]: same here.

File: SolastaCommunityExpansion/Patches/Camera/CameraControllerLocationPatcher.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+{
+    [HarmonyPatch(typeof(CameraControllerLocation), "FollowCharacterForBattle")]
+    [SuppressMessage("Minor Code Smell", "S101:Types should be named in PascalCase", Justification = "Patch")]
+    internal static class CameraControllerLocation_FollowCharacterForBattle
+    {
+        internal static bool Prefix(CameraControllerLocation __instance, GameLocationCharacter character)
+        {
+            try
+            {
+                if (!__instance) { Main.Log("ccl == null"); return true; }
```
Comment (ThyWoof) [inline SolastaCommunityExpansion/Patches/Camera/CameraControllerLocationPatcher.cs:16]: I guess our linting will complain about these if formats...

File: SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs (hunk 1 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs]: Fixed

File: SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs (hunk 2 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Viewers/Displays/GameUiDisplay.cs]: Fixed

File: SolastaCommunityExpansion/Patches/Camera/CameraControllerLocationPatcher.cs (hunk 3 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Patches/Camera/CameraControllerLocationPatcher.cs]: Tidied up.  (it didn't complain though)

**Review by ThyWoof — APPROVED** [review]

**Pull request merged**